packages/pigeon/lib/objc_generator.dart

```diff
@@ -953,47 +951,61 @@ static id GetNullableObjectAtIndex(NSArray* array, NSInteger key) {
       });
     }
 
-    void writeFromMap() {
-      indent.write('+ ($className *)fromMap:(NSDictionary *)dict ');
+    void writeFromList() {
+      indent.write('+ ($className *)fromList:(NSMutableArray *)list ');
       indent.scoped('{', '}', () {
         const String resultName = 'pigeonResult';
         indent.writeln('$className *$resultName = [[$className alloc] init];');
-        for (final NamedType field in klass.fields) {
+        final int len = klass.fields.length;
+        indent.scoped('if (list == nil || list == [NSNull null]) {', '}', () {
+          indent.writeln(
+              'list = [[NSMutableArray alloc] initWithCapacity: $len];');
+          indent.write('for (int i = 0; i < $len; i++) ');
+          indent.scoped('{', '}', () {
+            indent.writeln('[list addObject: [NSNull null]];');
+          });
+        });
+        klass.fields
+            .toList()
+            .asMap()
+            .forEach((int index, final NamedType field) {
           if (enumNames.contains(field.type.baseName)) {
             indent.writeln(
-                '$resultName.${field.name} = [${_dictGetter(classNames, 'dict', field, options.prefix)} integerValue];');
+                '$resultName.${field.name} = [(${_listGetter(classNames, 'list', field, index, options.prefix)} == [NSNull null] ? 0 : ${_listGetter(classNames, 'list', field, index, options.prefix)}) integerValue];');
           } else {
             indent.writeln(
-                '$resultName.${field.name} = ${_dictGetter(classNames, 'dict', field, options.prefix)};');
+                '$resultName.${field.name} = ${_listGetter(classNames, 'list', field, index, options.prefix)};');
```

Comment:
If `_listGetter` can return `NSNull`, then this is a type violation. I suspect given the above and this that `_listGetter` returning `NSNull` rather than `nil` is probably just not what we want.

Was the previous behavior that the dictionary only contained entries for non-null values? If so, then it was returning `nil`.